tests/zero_code_change/tensorflow2_gradtape_integration_tests.py

```diff
@@ -0,0 +1,169 @@
+Here in the test suite we delete the hook after every script.
+"""
+# Standard Library
+import argparse
+
+# Third Party
+import tensorflow.compat.v2 as tf
+
+# First Party
+import smdebug.tensorflow as smd
+from smdebug.core.utils import SagemakerSimulator
+
+
+def get_keras_data():
+    mnist = tf.keras.datasets.mnist
+    (x_train, y_train), (x_test, y_test) = mnist.load_data()
+
+    return (x_train, y_train), (x_test, y_test)
+
+
+def helper_test_keras_v2_gradienttape(script_mode: bool = False, json_file_contents="{}"):
+    """ Test the default ZCC behavior of saving losses and metrics in eager and non-eager modes."""
+    smd.del_hook()
+    tf.keras.backend.clear_session()
+
+    with SagemakerSimulator(json_file_contents=json_file_contents) as sim:
+        model = tf.keras.models.Sequential(
+            [
+                tf.keras.layers.Flatten(input_shape=(28, 28, 1)),  # WA for TF issue #36279
+                tf.keras.layers.Dense(128, activation="relu"),
+                tf.keras.layers.Dropout(0.2),
+                tf.keras.layers.Dense(10, activation="softmax"),
+            ]
+        )
+        (x_train, y_train), _ = get_keras_data()
+        dataset = tf.data.Dataset.from_tensor_slices(
+            (tf.cast(x_train[..., tf.newaxis] / 255, tf.float32), tf.cast(y_train, tf.int64))
+        )
+        dataset = dataset.shuffle(1000).batch(64)
+
+        opt = tf.keras.optimizers.RMSprop()
+        cce = tf.keras.losses.CategoricalCrossentropy(from_logits=True)
+        train_acc_metric = tf.keras.metrics.SparseCategoricalAccuracy()
+        n_epochs = 5
+        if script_mode:
+            if json_file_contents == "{}":
+                hook = smd.KerasHook(out_dir=sim.out_dir, export_tensorboard=True)
+            else:
+                hook = smd.KerasHook.create_from_json_file()
+
+            for epoch in range(n_epochs):
+                print("Epoch %d/%d" % (epoch + 1, n_epochs))
+                for data, labels in dataset:
+                    dataset_labels = labels
+                    labels = tf.one_hot(labels, depth=10)
+                    with hook.wrap_tape(tf.GradientTape()) as tape:
+                        logits = model(data, training=True)  # (32,10)
+                        loss_value = cce(labels, logits)
+                    grads = tape.gradient(loss_value, model.variables)
+                    opt.apply_gradients(zip(grads, model.variables))
+                    acc = train_acc_metric(dataset_labels, logits)
+                    hook.record_tensor_value(tensor_name="accuracy", tensor_value=acc)
+                log = "Epoch %d " % (epoch + 1)
+                log += "Accuracy %.4f" % train_acc_metric.result()
+                print(log)
+                train_acc_metric.reset_states()
+            hook = smd.get_hook()
+            assert hook
+            hook.close()
+            # Check that hook created and tensors saved
+            trial = smd.create_trial(path=sim.out_dir)
```

Comment:
Do we clean the `out_dir` get deleted in this test

Reply:
Okay, `SagemakerSimulator` cleans up the files.

Reply:
sagemaker simulator removes the existing out_dir at the beginning of the next test. 

```
class SagemakerSimulator(object):
    """
    Creates an environment variable pointing to a JSON config file, and creates the config file.
    Used for integration testing with zero-code-change.

    If `disable=True`, then we still create the `out_dir` directory, but ignore the config file.
    """

    def __init__(
        self,
        json_config_path="/tmp/zcc_config.json",
        tensorboard_dir="/tmp/tensorboard",
        training_job_name="sm_job",
        json_file_contents="{}",
        cleanup=True,
    ):
        self.out_dir = DEFAULT_SAGEMAKER_OUTDIR
        self.json_config_path = json_config_path
        self.tb_json_config_path = DEFAULT_SAGEMAKER_TENSORBOARD_PATH
        self.tensorboard_dir = tensorboard_dir
        self.training_job_name = training_job_name
        self.json_file_contents = json_file_contents
        self.cleanup = cleanup

    def __enter__(self):
        if self.cleanup is True:
            shutil.rmtree(self.out_dir, ignore_errors=True)
        
```